src/arrayMethodSort.js

```diff
@@ -4,8 +4,30 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  [].__proto__.sort2 = function(
+    compareFunction = (a, b) => String(a) < String(b)
+      ? -1
+      : 0
+  ) {
+    let countChanges = 0;
+
+    do {
+      countChanges = 0;
+
+      for (let i = 0; i < this.length - 1; i++) {
+        const currentItem = this[i];
+        const nextItem = this[i + 1];
+        const itemsComparetion = compareFunction(nextItem, currentItem);
+
+        if (itemsComparetion < 0) {
+          this[i] = nextItem;
+          this[i + 1] = currentItem;
+          countChanges++;
+        }
+      }
+    } while (countChanges > 0);
```

Comment:
counting changes and going through the whole array each time is not the best way to implement bubble sort